packages/trader/src/Modules/Trading/Components/Form/TradeParams/amount-mobile.tsx

```diff
@@ -159,6 +187,8 @@ const Amount = observer(
 
         return (
             <div>
+                {/* 
+                // @ts-expect-error We explicitly defined children as React.ReactElement[] here we only return a single JSX.Element or null. Dont know how to fix without breaking Tabs it's used in 27 files*/}
```

Comment:
@henry-deriv  instead of this @ts-expect-error, please try to apply the following suggestion for Line 228:
```
}) as React.ReactElement[]}
```

Reply:
}) as unknown as React.ReactElement[]}

Reply:
@henry-deriv  you will have to fix `TTabsProps` in `tabs.tsx` anyway because many non-required props there are marked as required.
The correct `children` type for Tabs component is the following:
```
children: (React.ReactElement | null)[];
```
After updating Tabs props, you will face some TS errors in this file that are easy to fix ("-" sign marks a removed line, "+" sign marks an added line):
```ts
diff --git a/packages/components/src/components/tabs/tabs.tsx b/packages/components/src/components/tabs/tabs.tsx
index c7d8f5483e..f0b4e3c3d0 100644
--- a/packages/components/src/components/tabs/tabs.tsx
+++ b/packages/components/src/components/tabs/tabs.tsx
@@ -14,20 +14,20 @@ declare module 'react' {
 }
 
 type TTabsProps = RouteComponentProps & {
-    active_icon_color: string;
+    active_icon_color?: string;
     active_index?: number;
-    background_color: string;
+    background_color?: string;
     bottom?: boolean;
-    center: boolean;
-    children: React.ReactElement[];
+    center?: boolean;
+    children: (React.ReactElement | null)[];
     className?: string;
-    fit_content: boolean;
+    fit_content?: boolean;
     has_active_line?: boolean;
     has_bottom_line?: boolean;
     header_fit_content?: boolean;
-    history: History;
-    icon_color: string;
-    icon_size: number;
+    history?: History;
+    icon_color?: string;
+    icon_size?: number;
     is_100vw?: boolean;
     is_full_width?: boolean;
     is_overflow_hidden?: boolean;
@@ -92,14 +92,14 @@ const Tabs = ({
         if (should_update_hash) {
             // if hash is in url, find which tab index correlates to it
             const hash = location.hash.slice(1);
-            const hash_index = children.findIndex(child => child && child.props && child.props.hash === hash);
-            const has_hash = hash_index > -1;
+            const hash_index = children?.findIndex(child => child && child.props && child.props.hash === hash);
+            const has_hash = Number(hash_index) > -1;
 
             if (has_hash) {
-                initial_index_to_show = hash_index;
+                initial_index_to_show = Number(hash_index);
             } else {
                 // if no hash is in url but component has passed hash prop, set hash of the tab shown
-                const child_props = children[initial_index_to_show].props;
+                const child_props = children?.[initial_index_to_show]?.props;
                 const current_id = child_props && child_props.hash;
                 if (current_id) {
                     pushHash(current_id);
@@ -128,19 +128,19 @@ const Tabs = ({
 
     const onClickTabItem = (index: number) => {
         if (should_update_hash) {
-            const hash = children[index].props['data-hash'];
+            const hash = children?.[index]?.props['data-hash'];
             pushHash(hash);
         }
         setActiveTabIndex(index);
         setActiveLineStyle();
     };
 
-    const valid_children = children.filter(child => child);
+    const valid_children = children?.filter(child => child);
 
     if (is_scrollable) {
         tab_width = 'unset';
     } else {
-        tab_width = fit_content ? '150px' : `${(100 / valid_children.length).toFixed(2)}%`;
+        tab_width = fit_content ? '150px' : `${(100 / Number(valid_children?.length)).toFixed(2)}%`;
     }
 
     return (
@@ -179,15 +179,15 @@ const Tabs = ({
                             const count = child.props['data-count'];
                             return (
                                 <Tab
-                                    active_icon_color={active_icon_color}
+                                    active_icon_color={active_icon_color ?? ''}
                                     className={className}
                                     count={count}
                                     icon={icon}
-                                    icon_color={icon_color}
-                                    icon_size={icon_size}
+                                    icon_color={icon_color ?? ''}
+                                    icon_size={icon_size ?? 0}
                                     is_active={index === active_tab_index}
                                     key={label}
-                                    is_label_hidden={children.length === 1 && single_tab_has_no_label}
+                                    is_label_hidden={children?.length === 1 && single_tab_has_no_label}
                                     label={label}
                                     id={id}
                                     is_scrollable={is_scrollable}
@@ -208,7 +208,8 @@ const Tabs = ({
                                     'dc-tabs__active-line--bottom': bottom,
                                     'dc-tabs__active-line--fit-content': fit_content,
                                     'dc-tabs__active-line--header-fit-content': header_fit_content,
-                                    'dc-tabs__active-line--is-hidden': children.length === 1 && single_tab_has_no_label,
+                                    'dc-tabs__active-line--is-hidden':
+                                        children?.length === 1 && single_tab_has_no_label,
                                 })}
                                 style={active_line_style}
                             />

```

Reply:
ok we wait until kates pr is merged